numba/unicode.py

```diff
@@ -451,6 +474,119 @@ def endswith_impl(a, b):
         return endswith_impl
 
 
+@overload_method(types.UnicodeType, 'split')
+def unicode_split(a, sep=None, maxsplit=-1):
+    if isinstance(sep, types.UnicodeType):
+        def split_impl(a, sep, maxsplit=-1):
+            a_len = len(a)
+            sep_len = len(sep)
+
+            if sep_len == 0:
+                raise ValueError('empty separator')
+
+            parts = []
+            last = 0
+            idx = 0
+            split_count = 0
+
+            while idx < a_len and (maxsplit == -1 or split_count < maxsplit):
+                if _cmp_region(a, idx, sep, 0, sep_len) == 0:
+                    parts.append(a[last:idx])
+                    idx += sep_len
+                    last = idx
+                    split_count += 1
+                else:
+                    idx += 1
+
+            if last <= a_len:
+                parts.append(a[last:])
+
+            return parts
+        return split_impl
+    elif sep is None:
+        def split_whitespace_impl(a):
```

Comment:
The `@overload` safety net needs extending to `@overload_method`. This function declaration has a signature that does not match the typing signature. `maxsplit` is not implemented and `sep` is missing as a `kwarg`, this breaks as a result:

```python
from numba import njit

def foo(x):
    return x.split(maxsplit=3)

args= ('\taa a a aa a aa a',)
print('"%s"' % njit(foo)(*args))
```

Reply:
fixed